kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -315,23 +316,45 @@ private static Predicate reverseComparatorFilter(Predicate predicate) {
     }
 
     /**
-     * Prunes the given schema according to the provided set of top-level leaf columns to keep. The
-     * columns in {@code schema} are taken to be nested under {@code parentPath}.
+     * Prunes the given schema to include only the referenced leaf columns to keep. These leaf
+     * columns (possible nested) are relative to the root schema, not to the current level of
+     * recursion. {@code leafColumnsToKeep} is unchanged at any level of recursion.
+     * <p>
+     * For example consider the following schema:
+     * <pre>
+     * |--level1_struct: struct
+     * |   |--level2_struct: struct
+     * |       |--level3_struct: struct
+     * |           |--level_4_col: int
+     * |       |--level_3_col: int
+     * </pre>
+     * At the second level of recursion on field {@code level2_struct} we would have parameters
+     * <ol>
+     *     <li>leafColumnsToKeep=Set(Column(level1_struct.level2_struct.level_3_col))</li>
+     *     <li>schema:
+     *          <pre>
+     *          |--level3_struct: struct
+     *          |   |--level_4_col: int
+     *          |--level_3_col: int
+     *          </pre>
+     *     </li>
+     *     <li>parentPath=["level1_struct", "level2_struct"]</li>
+     * </ol>
      *
      * @param schema schema to prune
-     * @param columnsToKeep set of top-level leaf columns to keep
+     * @param leafColumnsToKeep set of top-level leaf columns to keep
```

Comment:
I think `top-level leaf columns` is still confusing. I would just call it `leaf columns, relative to table schema root` or something along those lines

Really, since these are _exactly_ the same as what is passed to `pruneStatsSchema`, the same description should work and apply for both